src/attribution/onINP.ts

```diff
@@ -274,6 +280,10 @@ const attributeINP = (metric: INPMetric): INPMetricWithAttribution => {
   );
 
   const nextPaintTime = Math.max.apply(Math, nextPaintTimeCandidates);
+  // If processingEnd has been capped to inpTime then presentationDelay is 0
+  // Else use the nextPaintTime
+  const presentationDelay =
+    processingEnd == inpTime ? 0 : Math.max(nextPaintTime - processingEnd, 0);
```

Comment:
I think this is a fine temporary solution, but, consider just creating a variable such as "used_fallback_time", set up top when `endTime < processingEnd`, and re-use that everywhere.

(The goal is to expose a clue to event timing for this value, eventually.)

BTW the case where endTime is rounded to within 8ms of processingEnd is almost always another "used_fallback_time" example (i.e. when we actually dont have a paint)

Reply:
I'd prefer not to do that variable name. We haven't used a fallback time — Event timing has. We are always using the actual time given by that.

The current wording is similar to the LCP implementation in #527

Reply:
Thats fine.

---

The goal for Event Timing will be to:
- Expose `presentation_time` (possibly as `renderTime` to match other paint timing specs)
- Expose `fallback_time` (might be named something else)
- `duration` will be based on `fallback_time || presentation_time`, which is already is under the hood, but now you will know why.

Also, in hindsight, my suggestion of comparing with `processingEnd` only works if your event listener(s) caused the alert, but wouldn't work if some other task in the same animation frame did. Nor would it work for other cases of fallback_time like when you don't need next paint, or the page unloaded and hit visibilitychange as an alternative end time...